src/main/scala/com/mongodb/spark/rdd/partitioner/PartitionerHelper.scala

```diff
@@ -81,4 +84,72 @@ object PartitionerHelper {
     connector.withDatabaseDo(readConfig, { db => db.runCommand(collStatsCommand, readConfig.readPreference, classOf[BsonDocument]) })
   }
 
+  /**
+   * Returns the head `\$match` from a pipeline
+   *
+   * @param pipeline the users pipeline
+   * @return the head `\$match` or an empty `BsonDocument`.
+   */
+  def matchQuery(pipeline: Array[BsonDocument]): BsonDocument = {
+    val defaultQuery = new BsonDocument()
+    pipeline.headOption match {
+      case Some(document) => document.getDocument("$match", defaultQuery)
+      case None           => defaultQuery
+    }
+  }
+
+  /**
+   * Checks an aggregation pipeline to see if it starts with a range based query.
+   *
+   * If it does it returns the `\$match` filter otherwise None
+   *
+   * @param pipeline the aggregation pipeline
+   * @return the min and max keys for the pipeline
+   * @since 2.1
+   */
+  def getRangeFilterFromPipeline(partitionKey: String, pipeline: Array[BsonDocument]): (BsonValue, BsonValue) = {
+    val filter = pipeline.headOption match {
+      case Some(document: BsonDocument) => getNestedDocument(Seq("$match", partitionKey), document)
+      case None                         => new BsonDocument()
+    }
+    (filter.get("$gte", new BsonMinKey()), filter.get("$lt", new BsonMaxKey()))
```

Comment:
On review - this is because chunk ranges produced by spitVector match `$gte` and `$lt` semantics for the chunks. I'll add a comment in the code to clarify.